modules/personalize_taxonomy_context/plugins/visitor_context/TaxonomyContext.inc

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @file
+ * Provides a visitor context plugin for taxonomy.
+
+ */
+
+class TaxonomyContext extends PersonalizeContextBase {
+  /**
+   * Implements PersonalizeContextInterface::create().
+   */
+  public static function create(PersonalizeAgentInterface $agent = NULL, $selected_context = array()) {
+    return new self($agent, $selected_context);
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getOptions().
+   */
+  public static function getOptions() {
+    $vocabularies = taxonomy_get_vocabularies();
+    $vocabulary_names = array();
+    foreach ($vocabularies as $vid => $vocabulary) {
+      $vocabulary_names[$vid] = $vocabulary->name;
+    }
+
+    $options = array();
+    foreach ($vocabulary_names as $vid => $name) {
+      $options[$vid] = array(
+        'name' => $name,
+        'group' => 'Taxonomy',
+      );
+    }
+
+    return $options;
+  }
+
+  /**
+   * Implements PersonalizeContextInterface::getPossibleValues().
+   */
+  public function getPossibleValues($limit = FALSE) {
+    $possible_values = array();
+    $options = $this->getOptions();
+    foreach ($options as $name => $info) {
+      $possible_values[$name] = array(
+        'value type' => 'string',
```

Comment:
We need to get all the actual possible values here (unless it's a free-tagging vocabulary)